app/src/main/java/com/example/event_finder_capstone/EventDetailsActivity.java

```diff
@@ -0,0 +1,113 @@
+    TextView tvEventDetailsSite;
+    TextView tvEventDetailsCost;
+    ImageView ivEventDetailsImage;
+    Button btnPhotoAlbum;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_details);
+
+        tvEventDetailsTitle = findViewById(R.id.tvEventDetailsTitle);
+        tvEventDetailsDescription = findViewById(R.id.tvEventDetailsDescription);
+        tvEventDetailsStartDate = findViewById(R.id.tvEventDetailsStartDate);
+        tvEventDetailsEndDate = findViewById(R.id.tvEventDetailsEndDate);
+        tvEventDetailsAddress = findViewById(R.id.tvEventDetailsAddress);
+        tvEventDetailsSite = findViewById(R.id.tvEventDetailsSite);
+        tvEventDetailsCost = findViewById(R.id.tvEventDetailsCost);
+        ivEventDetailsImage = findViewById(R.id.ivEventsDetailsImage);
+        btnPhotoAlbum = findViewById(R.id.btnPhotoAlbum);
+
+        setDetailsScreenText();
+
+        btnPhotoAlbum.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                goPhotoAlbum();
+            }
+        });
+    }
+
+    private void goPhotoAlbum() {
+        Intent intent = new Intent(EventDetailsActivity.this, PhotoAlbumActivity.class);
+        startActivity(intent);
+    }
+
+    private void setDetailsScreenText() {
+        event = Parcels.unwrap(getIntent().getParcelableExtra(Event.class.getSimpleName()));
+
+        tvEventDetailsTitle.setText(event.getName());
+        tvEventDetailsDescription.setText(event.getDescription());
+        formatAndSetEventURL();
+        tvEventDetailsAddress.setText(event.getLocation());
+        tvEventDetailsCost.setText(event.getCost());
+        Glide.with(this).load(event.getImageUrl()).centerCrop().into(ivEventDetailsImage);
+        setEventStartAndEndDates();
+    }
+
+    private void setEventStartAndEndDates() {
+        try {
+            tvEventDetailsStartDate.setText(convertEventDateFormat(event.getTimeStart()));
+        } catch (ParseException e) {
+            Toast.makeText(EventDetailsActivity.this, "Failed to retrieve start date", Toast.LENGTH_SHORT).show();
+        }
+        try {
+            tvEventDetailsEndDate.setText(convertEventDateFormat(event.getTimeEnd()));
+        } catch (ParseException e) {
+            Toast.makeText(EventDetailsActivity.this, "Failed to retrieve end date", Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    private void formatAndSetEventURL() {
+        String link = event.getEventSiteUrl();
+        tvEventDetailsSite.setMovementMethod(LinkMovementMethod.getInstance());
+        tvEventDetailsSite.setClickable(true);
+        String text = "<a href='" + link + "'> Book Tickets and Learn More </a>";
+        tvEventDetailsSite.setText(Html.fromHtml(text));
+    }
+
+    private String convertEventDateFormat(String unformattedDate) throws ParseException {
```

Comment:
🛠️  This appears to be the same formatting as what's in `EventsAdapter`.  Consider how this can be refactored to be reusable between classes.